File: backend/app/schemas/recommendation.py
from pydantic import BaseModel, ConfigDict

from app.schemas.opportunity import OpportunityOut


class ScoreBreakdownOut(BaseModel):
    skill_score: float
    eligibility_score: float
    interest_score: float
    deadline_score: float
    mode_score: float


class RecommendationOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    opportunity: OpportunityOut
    match_score: float
    eligibility_status: str
    reasons: list[str]
    score_breakdown: ScoreBreakdownOut

File: frontend/src/types/recommendation.ts
import type { Opportunity } from "./opportunity";

export interface ScoreBreakdown {
  skill_score: number;
  eligibility_score: number;
  interest_score: number;
  deadline_score: number;
  mode_score: number;
}

export interface Recommendation {
  opportunity: Opportunity;
  match_score: number;
  eligibility_status: "ELIGIBLE" | "NOT_ELIGIBLE" | "UNCERTAIN";
  reasons: string[];
  score_breakdown: ScoreBreakdown;
}